runtime/compiler/z/codegen/S390PrivateLinkage.cpp

```diff
@@ -2592,7 +2699,9 @@ J9::Z::PrivateLinkage::buildDirectCall(TR::Node * callNode, TR::SymbolReference
 
    if (comp()->getOption(TR_TraceCG))
       traceMsg(comp(), "Build Direct Call\n");
-
+   // TODO: For read only code cache, we will need to change the jitInduceOSR
+   // call below to use the sequence that is works for RXi, Probably a new
+   // buildDirectCall makes sense here to handle all the cases more cleanly
    // generate call
```

Comment:
I have added a FATAL assert with Node in case generateReadOnlyCode is enabled and we encounter jitInduceOSR call. 
https://github.com/eclipse/openj9/compare/9a9d9b37daeb983c8d4e295fcc785afeecde5d78..de725983cd06ff96468a76b30321d3f8bb48caea